azurerm/internal/services/web/app_service.go

```diff
@@ -1608,7 +1608,9 @@ func expandAppServiceSiteConfig(input interface{}) (*web.SiteConfig, error) {
 
 		documents := make([]string, 0)
 		for _, document := range input {
-			documents = append(documents, document.(string))
+			if document != nil {
+				documents = append(documents, document.(string))
+			}
```

Comment:
This will skip the value, returning `[]` rather than `[""]` meaning it will not behave the same as if `split(",", "")` had returned correctly.